test/functionalTest/cases/1492_limit_zero/limit_zero_entities.test

```diff
@@ -0,0 +1,160 @@
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
+# VALGRIND_READY - to mark the test ready for valgrindTestSuite.sh
+
+--NAME--
+limit_zero_entities
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB
+
+--SHELL--
+
+#
+# 01. GET /v2/entities with 'limit' == 0 and options=count and see no entities and fiware-total-count: 0
+# 02. POST /v2/entities to create E1 entity
+# 03. POST /v2/entities to create E1 entity
+# 04. GET /v2/entities with 'limit' == 0 and options=count and see no entities and fiware-total-count: 2
+# 05. GET /v2/entities with 'limit' == 0 and see no entities
+#
+
+echo "01. GET /v2/entities with 'limit' == 0 and options=count and see no entities and fiware-total-count: 0"
+echo "======================================================================================================"
+orionCurl --url "/v2/entities?limit=0&options=count"
+echo
+echo
+
+
+echo "02. POST /v2/entities to create E1 entity"
+echo "========================================="
+payload='{
+  "id": "E1",
+  "type": "T"
+}'
+orionCurl --url /v2/entities -X POST --payload "$payload"
+echo
+echo
+
+
+echo "03. POST /v2/entities to create E2 entity"
+echo "========================================="
+payload='{
+  "id": "E2",
+  "type": "T"
+}'
+orionCurl --url /v2/entities -X POST --payload "$payload"
+echo
+echo
+
+
+echo "04. GET /v2/entities with 'limit' == 0 and options=count and see no entities and fiware-total-count: 2"
+echo "======================================================================================================"
+orionCurl --url "/v2/entities?limit=0&options=count"
+echo
+echo
+
+
+echo "05. GET /v2/entities with 'limit' == 0 and see no entities"
+echo "=========================================================="
+orionCurl --url "/v2/entities?limit=0"
+echo
+echo
+
+
+
+--REGEXPECT--
+01. GET /v2/entities with 'limit' == 0 and options=count and see no entities and fiware-total-count: 0
+======================================================================================================
+HTTP/1.1 200 OK
+Content-Length: 2
+Content-Type: application/json
+Fiware-Total-Count: 0
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+[]
+
+
+02. POST /v2/entities to create E1 entity
+=========================================
+HTTP/1.1 201 Created
+Content-Length: 0
+Location: /v2/entities/E1?type=T
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+
+
+03. POST /v2/entities to create E2 entity
+=========================================
+HTTP/1.1 201 Created
+Content-Length: 0
+Location: /v2/entities/E2?type=T
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+
+
+04. GET /v2/entities with 'limit' == 0 and options=count and see no entities and fiware-total-count: 2
```

Comment:
This step (and also step 5) are not working correctly. If you use `limit=0` then no entity should be returned but two ones are returned in this case.

Thus, maybe it is not enough with modifying the parameter checking logic in src/lib/rest/rest.cpp and another modifications are needed.

Reply:
A similar situation is happending also in limit_zero_subs.test and limit_zero_regs.test

Reply:
> This step (and also step 5) are not working correctly. If you use `limit=0` then no entity should be returned but two ones are returned in this case.
> 
> Thus, maybe it is not enough with modifying the parameter checking logic in src/lib/rest/rest.cpp and another modifications are needed.

Done modification in https://github.com/telefonicaid/fiware-orion/pull/3994/commits/cf68c96ee05fa0fd7b53156f8e2b86761a386d0f It works fine in case of limit_zero_subs.test and limit_zero_regs.test but in case of entity it doesn't show fiware-total-count in response. Also I haven't find code for count in getEntities() in src/lib/serviceRoutinesV2/getEntities.cpp. Can you please suggest?

Reply:
> Also I haven't find code for count in getEntities() in src/lib/serviceRoutinesV2/getEntities.cpp. Can you please suggest?

If you search for HTTP_FIWARE_TOTAL_COUNT in the code base you would find the places in which this header is being used so you can see how the calculation is done in each case:

![imagen](https://user-images.githubusercontent.com/1534240/143000149-bd7324e8-8e00-4f67-83b6-85916ca7cac9.png)

In the case of entities I'd say that the first one (postQueryContext.cpp) is the one to look for.

As additional comment, looking to the above results, I'd realized that the `GET /v2/types` is also paginated, so a limit_zero_types.test variants should be also added to this PR.

Reply:
Added test case for types in https://github.com/telefonicaid/fiware-orion/pull/3994/commits/64cf793d7fae599dcef70fc063ce6302eeb9e5bd